kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/DMNCompilerImpl.java

```diff
@@ -449,27 +449,11 @@ private DMNExpressionEvaluator compileLiteralExpression(DMNBaseNode node, Litera
         return evaluator;
     }
 
-    /**
-     * TODO quick hack to parse values, in case they are a list
-     * @param text
-     * @return
-     */
-    protected static List<UnaryTest> textToUnaryTestList(String text) {
+    protected static List<UnaryTest> textToUnaryTestList(FEEL feel, String text) {
         if (text == null || text.isEmpty()) {
             return Collections.emptyList();
         }
-        List<Object> ie = (List<Object>) FEEL.newInstance().evaluate( "[ " + text + " ]" );
-        List<UnaryTest> tests = new ArrayList<>(  );
-        for( Object o : ie ) {
-            if ( o instanceof UnaryTest ) {
-                tests.add( (UnaryTest) o );
-            } else if ( o instanceof Range ) {
-                tests.add( (c, x) -> x != null && ((Range) o).includes( (Comparable<?>) x ) );
-            } else {
-                tests.add( (c, x) -> x != null && x.equals( o ) );
-            }
-        }
-        return tests;
+        return feel.evaluateUnaryTests( text );
```

Comment:
Defer variables evaluation until runtime; we do so by rewriting this node by prepending with `=` as needed if it's a variable.

This is required, otherwise such variables would be evaluated at runtime - paradox as their value will be known only at runtime.